### PR TITLE
Zephyr Memory Management FW Architecture

### PR DESCRIPTION
**Pull request by mwasko**
Memory management description, design and flows based on Zephyr infrastructure

### CODE REVIEW

File: architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst [hunk 1]
```diff
@@ -3,14 +3,55 @@ Heap sharing
...
-The data structures needed to manage shared memories are initialized by primary
-core, structure location in memory map is known at the build time and access to
-it is controlled by mutex.
+.. note:: Introduction of MMU will require a separate local application heap per
+   isolated domain.
+
+L1 Cache Coherency
+******************
+
+NOTE: This section applies to systems without L1 cache coherency.
```
Comment (lgirdwood) [inline architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst:18]: We probably want to say on Intel architecture only atm, I'm sure others will need this advice too for multicore on xtensa.

File: architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst [hunk 2]
```diff
@@ -3,14 +3,55 @@ Heap sharing
...
+
+Option #1 is preferred, since does not require explicit L1 cache operations
+when memory is accessed by a DSP core. However all operations directly access
+L2+ memory therefore it is not suitable for a low latency high performance data
+processing case.
+
+Option #2 provides better performance but requires explicit L1 cache operations,
+which are difficult to maintain and validate, as well as data cache alignment
+for both client buffers and their descriptors, which creates an overhead. This
+configuration should be avoided if possible.
```
Comment (lgirdwood) [inline architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst:41]: but it is possible to share data across core L1 cache via the coherent API.

**Review by lgirdwood — COMMENTED** [review]
I think we need to also state when (SOF and or Zephyr version) each feature or API is upstream. Long term we should link each section to the relevant doxygen pages.

**Comment by lgirdwood** [comment]
> @lgirdwood Will this be part of the 2.4.1 release?

No, not needed until next release.

File: architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst (hunk 1 above)
Comment (mwasko) [reply architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst]: update the note that it applies to Intel arch

File: architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst (hunk 2 above)
Comment (mwasko) [reply architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst]: I have added this info to the description

**Review by mmaka1 — APPROVED** [review]

**Comment by mwasko** [comment]
@lgirdwood are you good with the updated version?

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**